xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/Inputstream.h

```diff
@@ -0,0 +1,585 @@
+#pragma once
+/*
+ *      Copyright (C) 2005-2017 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Kodi; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/*
+ * Parts with a comment named "internal" are only used inside header and not
+ * used or accessed direct during add-on development!
+ */
+
+#include "../AddonBase.h"
+
+#ifdef BUILD_KODI_ADDON
+#include "../DVDDemuxPacket.h"
+#else
+#include "cores/VideoPlayer/DVDDemuxers/DVDDemuxPacket.h"
+#endif
+
+namespace kodi { namespace addon { class CInstanceInputStream; }}
+
+extern "C" {
+
+  /*!
+   * @brief InputStream add-on capabilities. All capabilities are set to "false" as default.
+   */
+  typedef struct INPUTSTREAM_CAPABILITIES
+  {
+    enum MASKTYPE: uint32_t
+    {
+      /// supports interface IDemux
+      SUPPORTSIDEMUX = (1 << 0),
+
+      /// supports interface IPosTime
+      SUPPORTSIPOSTIME = (1 << 1),
+
+      /// supports interface IDisplayTime
+      SUPPORTSIDISPLAYTIME = (1 << 2),
+
+      /// supports seek
+      SUPPORTSSEEK = (1 << 3),
+
+      /// supports pause
+      SUPPORTSPAUSE = (1 << 4)
```

Comment:
Maybe it's just personal style, but I would insert an underline between the verb and the action (e.g. SUPPORTS_PAUSE). Increases readability and this is how we usuallly name enum members, iirc.